File: packages/java/anip-core/src/main/java/dev/anip/core/CheckpointListResponse.java
package dev.anip.core;

import java.util.List;

/**
 * Response from the checkpoint list endpoint.
 */
public class CheckpointListResponse {

    private final List<Checkpoint> checkpoints;
    private final String nextCursor;

    public CheckpointListResponse(List<Checkpoint> checkpoints, String nextCursor) {
        this.checkpoints = checkpoints;
        this.nextCursor = nextCursor;
    }

    public List<Checkpoint> getCheckpoints() {
        return checkpoints;
    }

    public String getNextCursor() {
        return nextCursor;
    }
}

File: packages/csharp/src/Anip.Core/CheckpointListResponse.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class CheckpointListResponse
{
    [JsonPropertyName("checkpoints")]
    public List<Checkpoint> Checkpoints { get; set; } = new();

    [JsonPropertyName("next_cursor")]
    public string? NextCursor { get; set; }
}
